Piloted and revised Foundational HPC course (#187)
UNIVERSE-HPC/course-material: high_performance_computing/computer_simulations/01_intro.md

© Modelling Commons - 1997 Uri Wilensky

---

![Overhead photo of runners on a track](images/steven-lelham-atSaEOeE8Nk-unsplash.jpg)
*Image courtesy of [Steven Lelham](https://unsplash.com/@slelham) from [Unsplash](https://unsplash.com)*

## Wolf-sheep predation simulation - Initial and Boundary Conditions

Other concepts that we have not talked about yet, are boundary and initial condition. In order to discuss these concepts we would like you to explore how they affect the wolf-sheep predation model.

In this case the boundary conditions define what happens to wolves and sheep that get to the edge of the area covered by the simulation. Try to stop the simulation when one of the animals is about to go out of the frame. Did you notice that it appears immediately on the opposite side of the frame? This behaviour is due to the periodic boundary conditions.

As you know, initial conditions refer to the parameters you choose before running the simulation e.g. the initial number of wolves and sheep, the grass regrow period and the reproduction rates of both populations.

> To understand the importance of choosing the right input parameters, and see the difference even the smallest variation can produce, you are encouraged to play with the values of different parameters. Try to find values of parameters that would:
> 
> - allow both populations to be stable (i.e. neither population is dying out) even after 1000 time steps
> - allow the sheep population to die out
> - allow the wolf population to die out

Reviewer: ```suggestion
To understand the importance of choosing the right input parameters, and see the difference even the smallest variation can produce, you are encouraged to play with the values of different parameters.
Try to find values of parameters that:

- Allow both populations to be stable (i.e. neither population is dying out) even after 1000 time steps.
- Allow the sheep population to die out.
- Allow the wolf population to die out.
```

> How difficult was it to find the sets of parameters to meet the conditions? Have you noticed anything unexpected?

Reviewer: ```suggestion
:::callout{variant="discussion}
How difficult was it to find the sets of parameters to meet the conditions? Have you noticed anything unexpected?
:::
```

## Computational Science

Computational science is a rapidly growing interdisciplinary field. There are many problems in science and technology that cannot be sufficiently studied experimentally or theoretically. It may be too expensive or too dangerous, or simply impossible due to the space and timescales involved.

In fact, computational science is considered by many to be a third methodology in scientific research, along with theory and experiment, and working in tandem with them. Computational science can be used to corroborate theories that cannot be confirmed or denied experimentally, for example theories relating to the creation of the universe. On the other hand, advances in experimental techniques and the resulting data explosion, allow for data-driven modelling and simulation.

You should not confuse computational science, which uses computational methods to deal with scientific problems, and computer science, which focuses on the computer itself. Having said that, computational science draws upon computer science, as well as upon mathematics and applied sciences. Computational science typically consists of three main components:

> 1) algorithms and models
> 2) software developed to solve problems and
> 3) the computer and information infrastructure e.g. hardware, networking and data management components.

Reviewer: ```suggestion
1. algorithms and models
1. software developed to solve problems and
1. the computer and information infrastructure e.g. hardware, networking and data management components.
```

Clearly, computational science is an intersection between mathematics, applied disciplines and computer science.

> Some of the disciplines traditionally associated with computational science include: atmospheric sciences (e.g. weather forecasting, climate and ocean modelling, seismology etc.), astrophysics, nuclear engineering, chemistry, fluid dynamics, structural analysis and material research. It’s easy to see why these disciplines were quick to take up computational science.
> 
> However, more recently, other disciplines, such as medicine (e.g. medical imaging, blood flow simulations, bone structure simulations), biology (e.g. ecosystem and environmental modelling) and economics became part of computational science as well. It is quite common nowadays to come across terms like computational economics or computational biology.

Reviewer: ```suggestion
Some of the disciplines traditionally associated with computational science include: atmospheric sciences (e.g. weather forecasting, climate and ocean modelling, seismology etc.), astrophysics, nuclear engineering, chemistry, fluid dynamics, structural analysis and material research. It’s easy to see why these disciplines were quick to take up computational science.

Other disciplines, such as medicine (e.g. medical imaging, blood flow simulations, bone structure simulations), biology (e.g. ecosystem and environmental modelling) and economics are embracing computational science as well.
It has become quite common to come across terms like computational economics or computational biology.

Even more recently, advances in machine learning has led to an entirely new subfield of computational science and its application across a huge range of disciplines.
```

> If you want to know what sorts of applications are run on the UK’s national supercomputer ARCHER2, visit the relevant page on the ARCHER2 website. Is there anything that surprises you there?

Reviewer: ```suggestion
:::callout{variant="discussion"}
If you want to know what sorts of applications are run on the UK’s national supercomputer ARCHER2, visit the relevant page on the ARCHER2 website. Is there anything that surprises you there?
:::
```

---

![Question marks](images/laurin-steffens-IVGZ6NsmyBI-unsplash.jpg)
*Image courtesy of [Laurin Steffens](https://unsplash.com/@lausteff) from [Unsplash](https://unsplash.com)*

## Why use supercomputers?

> Having learned how computer simulations work and what supercomputers are capable of you should have a good idea why we need them.
> 

Reviewer: ```suggestion
```
